packages/runtime-core/src/componentPublicInstance.ts

```diff
@@ -379,9 +379,7 @@ export const PublicInstanceProxyHandlers: ProxyHandler<any> = {
           return desc.get.call(instance.proxy)
         } else {
           const val = globalProperties[key]
-          return isFunction(val)
-            ? Object.assign(val.bind(instance.proxy), val)
-            : val
+          return isFunction(val) ? extend(val.bind(instance.proxy), val) : val
```

Comment:
`extend` is an alias of `Object.assign`, its main purpose is reduced minified size for the runtime, and avoid user build tools from transpiling `Object.assign` to something inefficient. In other cases like `compiler-sfc` and `template-explorer`, it's not necessary. This is the only place where it is justified.

Reply:
I have already made the modifications according to your suggestions. In the future, I will pay attention to these issues. Thank you for your guidance.